### ISSUE TITLE
maximum/minimum give a tie's whole gradient to one operand, so BCEWithLogitsLoss is wrong at a logit of 0

### ISSUE THREAD

**Issue by vyncint** [vyncint/oxmera#84]
## Summary

The VJP of the elementwise `maximum`/`minimum` uses a **strict** comparison, so on a tie the entire gradient goes to one operand. `BCEWithLogitsLoss` is built out of `maximum(x, 0)`, so at a logit of exactly `0.0` it returns a wrong gradient — and gradient descent moves *uphill*.

The per-element derivative of BCE-with-logits is exactly `sigmoid(x) - t`, which at `x = 0` is `0.5 - t`. The forward value is correct (`ln 2`), so nothing signals the problem.

This also falsifies a documented promise: `docs/LIMITATIONS.md` states the convention is that "`max` ties split evenly", which is precisely the behaviour that would make this loss exact at 0.

Logit `0.0` is routine, not contrived: `Linear` initialises bias to zero, so a zero/masked/padded input row, a zero-initialised head, or a dead ReLU stack all produce it on the first steps.

## Reproduction

```rust
// oxmera = "0.5.0"
use oxmera::nn::BCEWithLogitsLoss;
use oxmera::{Result, Tensor};

fn main() -> Result<()> {
    let logits = Tensor::from_slice(&[0.0, 0.0, 0.0, 0.0], [4])?.requires_grad_(true);
    let target = Tensor::from_slice(&[1.0, 1.0, 0.0, 0.0], [4])?;
    let loss = BCEWithLogitsLoss.forward(&logits, &target)?;
    loss.backward()?;
    println!("grad = {:?}", logits.grad().unwrap().to_vec_f32()?);
    Ok(())
}
```

Root cause, same crate:

```rust
let x = Tensor::from_slice(&[0.0, 1.0, -1.0], [3])?.requires_grad_(true);
let z = Tensor::from_slice(&[0.0, 0.0, 0.0], [3])?;
x.maximum(&z)?.sum(&[])?.backward()?;   // -> [0.0, 1.0, 0.0]
```

## Observed

```
grad = [-0.25, -0.25, 0.0, 0.0]
d/dx max(x,0) at x=[0,1,-1] = [0.0, 1.0, 0.0]
```

A central finite difference of oxmera's own forward gives `-0.49999952` per element where autograd returns `-1`. Sweeping across zero, the gradient tracks the exact value to 8 digits at `x = ±1e-6` and then jumps discontinuously at exactly `0`.

End to end, on a balanced batch where the true bias gradient is `0`, oxmera reports `-0.5`; one SGD step raises the loss from `0.6931472` to `0.724077`.

## Expected

`grad = [-0.125, -0.125, 0.125, 0.125]` — `(sigmoid(x) - t) / n`, which is what finite differences of oxmera's own forward give, what the closed form gives, and what PyTorch's fused `binary_cross_entropy_with_logits` backward returns.

Either give `BinaryOp::Maximum`'s VJP the evenly-split tie that `docs/LIMITATIONS.md` already promises, or give `BCEWithLogitsLoss` a direct `sigmoid(x) - t` VJP instead of composing it from `maximum`/`abs`.

## Where

- `crates/oxmera-tensor/src/ops.rs` — `BinaryOp::Maximum` VJP, strict `gt_mask`
- `crates/oxmera-nn/src/loss.rs` — `relu_x = logits.maximum(&zero)`
- `docs/LIMITATIONS.md` — the "`max` ties split evenly" claim

No loss is gradient-checked in the suite today, which is why this survived.
